x-pack/plugins/infra/public/components/asset_details/components/metadata_explanation.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiText, EuiLink } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { useDateRangeProviderContext } from '../hooks/use_date_range';
+import { Popover } from '../tabs/common/popover';
+
+const HOSTNAME_DOCS_LINK =
+  'https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-name';
+
+const MetadataExplanationTooltipContent = React.memo(() => {
+  const onClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    e.stopPropagation();
+  };
+
+  return (
+    <EuiText size="s" onClick={onClick} style={{ width: 200 }}>
+      <FormattedMessage
+        id="xpack.infra.assetDetails.metadata.tooltip.documentationLabel"
+        defaultMessage="{metadata} is populated from the last event detected for this {hostName} for the selected date period."
+        values={{
+          metadata: (
+            <i>
+              <FormattedMessage
+                id="xpack.infra.assetDetails.metadata.tooltip.metadata"
+                defaultMessage="Metadata"
+              />
+            </i>
+          ),
+          hostName: (
+            <EuiLink
+              data-test-subj="infraAssetDetailsTooltipDocumentationLink"
+              href={HOSTNAME_DOCS_LINK}
+              target="_blank"
+            >
+              <FormattedMessage
+                id="xpack.infra.assetDetails.metadata.tooltip.documentationLink"
+                defaultMessage="host.name"
+              />
+            </EuiLink>
+          ),
+        }}
+      />
+    </EuiText>
+  );
+});
+
+export const MetadataExplanationMessage = () => {
+  const { getDateRangeInTimestamp } = useDateRangeProviderContext();
+  const dateFromRange = new Date(getDateRangeInTimestamp().to);
+  const dateString = dateFromRange.toLocaleDateString();
+  const timeString = dateFromRange.toLocaleTimeString();
```

Comment:
This is OK (I think), but it's not entirely true. The true timestamp of the latest metadata info, should come from the `/api/infra/metadata` endpoint. 

The query is [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/server/routes/metadata/lib/get_node_info.ts#L62), adding `TIMESTAMP_FIELD` to `_source` and propagating it in the endpoint response will give the FE the right information.

Both values (metadata API timestamp and date picker timestamp) will be pretty close if the period in metricbeat is set to 10s, but I suspect that they might diverge if the period in metricbeat is set to 1m. 

Might be worth testing it.

Reply:
I am not sure if I understand this issue: I checked the request and the date/time shown in there and we have the `to` matching with the date shown: 
<img width="1738" alt="image" src="https://github.com/elastic/kibana/assets/14139027/c1d5541b-b19f-44ee-819b-29a17db09010">
In my metricbeat config I have: 
```
  - module: system
    metricsets:
        ............
    period: 1m
    processes: [".*"]
```
So I saw that we pass that time range when we do the request to get the metadata as this is the `to` date should we have "before" {date/time} not "on" {date/time} in the text to be clear as it's at the end of the range 🤔

Reply:
What you're seeing in your screenshot is the request payload? We pass that to the API, which runs a query using that range and returns that top 1 desc document. 

This top 1 desc doc timestamp could be the same or pretty close to `to`. But if the last document ingested was 30 seconds before `to`, then there should be a small difference between that and `to`. 

Not sure if it's clear. Happy to discuss over slack.